recipes/xerces-c/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        # https://xerces.apache.org/xerces-c/build-3.html
+        # TODO : check if we need options for that variants
+        cmake.definitions["network-accessor"] = {"Windows": "winsock",
+                                                 "Macos": "cfurl",
+                                                 "Linux": "socket"}.get(str(self.settings.os))
+        cmake.definitions["transcoder"] = {"Windows": "windows",
+                                           "Macos": "macosunicodeconverter",
+                                           "Linux": "gnuiconv"}.get(str(self.settings.os))
+        cmake.definitions["message-loader"] = "inmemory"
+        cmake.definitions["xmlch-type"] = "uint16_t"
+        cmake.definitions["mutex-manager"] = {"Windows": "windows",
+                                              "Macos": "posix",
+                                              "Linux": "posix"}.get(str(self.settings.os))
+        # avoid picking up system dependency
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_CURL'] = True
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_ICU'] = True
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        # remove unneeded directories
+        tools.rmdir(os.path.join(self.package_folder, 'share'))
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
+        tools.rmdir(os.path.join(self.package_folder, 'cmake'))
+
+    def package_info(self):
+        version_tokens = self.version.split(".")
+        if self.settings.os == "Windows":
+            lib = "xerces-c_%s" % version_tokens[0]
+            if self.settings.build_type == "Debug":
+                lib += "d"
+            self.cpp_info.libs = [lib]
+        else:
+            self.cpp_info.libs = ["xerces-c" if self.options.shared else
+                                  ("xerces-c-%s.%s" % (version_tokens[0], version_tokens[1]))]
+        if self.settings.os == "Macos":
+            frameworks = ['CoreFoundation', 'CoreServices']
+            for framework in frameworks:
+                self.cpp_info.exelinkflags.append("-framework %s" % framework)
```

Comment:
use `self.cpp_info.frameworks`

Reply:
Added, now it looks much nicer

Reply:
Hmm, does not work as expected.

Linking test_package before change (ignore the xcode version, I picked up a random one):
```
/Applications/Xcode-11.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++    -m64 -stdlib=libc++ -O3 -DNDEBUG  -isysroot /Applications/Xcode-11.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  -framework CoreFoundation -framework CoreServices    CMakeFiles/test_package.dir/GraphHandler.cpp.o 
```

Linking test_package after change:
```
/usr/local/Cellar/cmake/3.14.1/bin/cmake -E cmake_link_script CMakeFiles/test_package.dir/link.txt --verbose=1
/Applications/Xcode-9.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++    -m64 -stdlib=libc++ -g  -Wl,-search_paths_first -Wl,-headerpad_max_install_names      CMakeFiles/test_package.dir/test_package.cpp.o  -o bin/test_package  
```

Options `-framework` are missing despite we provided `self.cpp_info.frameworks`. Are we sure the latest 1.19.0 Conan is used?

Reply:
@theirix thanks, I'll check it

Reply:
seems to be working for me:
```
[100%] Linking CXX executable bin/test_package
/usr/local/Cellar/cmake/3.15.2/bin/cmake -E cmake_link_script CMakeFiles/test_package.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++    -m64 -stdlib=libc++ -O3 -DNDEBUG  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names      CMakeFiles/test_package.dir/test_package.cpp.o  -o bin/test_package  -L/Users/sse4/.conan/data/xerces-c/3.2.2/bincrafters/testing/package/eec6acc43f6348a597c20e5bd28d9e0590a02597/lib -lxerces-c-3.2 -framework CoreFoundation -framework CoreServices
```

Reply:
It works for me too on my local machine. The problem was reproduced only on CCI build machine. Let's try to see if removing `sharedlinkflags` helps.